File: JAVA/Largest Number formed from an Array.java
class Solution {
    String printLargest(int n, String[] arr) {
        Arrays.sort(arr, new Comparator<String>(){
            @Override
            public int compare(String x, String y){
                String xy=x+y;
                String yx= y+x;
                return yx.compareTo(xy);
            }
        });
        String ans="";
        for(int u=0;u<n;u++){
            ans+=arr[u];
        }
        return ans;
    }
}

File: C++/Largest Number formed from an Array.cpp
class Solution{
public:
   static bool cmp(string a,string b){
       return a+b>b+a;
   }
    string printLargest(int n, vector<string> &arr) {
        sort(arr.begin(),arr.end(),cmp);
        string ans="";
        for(int i=0;i<n;i++){
            ans+=arr[i];
        }
        while(ans[0]=='0' && ans.length()>1) ans.erase(0,1);
        return ans;
    }
};
